doc/src/contracts/multisig_prover.md

```diff
@@ -62,17 +63,101 @@ Prover-->>-Relayer: returns GetProofResponse
+	Signers->>+Multisig: signature collection
+  end
+Multisig-->>-Relayer: emit SigningCompleted event
+Relayer->>+Prover: QueryMsg::GetProof
+Prover->>+Multisig: QueryMsg::GetSigningSession
+Multisig-->>-Prover: reply with status, current signatures vector and snapshot
+Prover-->>-Relayer: returns GetProofResponse
+end
+```
+
+1. The Relayer calls Prover to update the `WorkerSet`.
+2. Replaces the current `WorkerSet` by saving the new `WorkerSet`.
+3. The new `WorkerSet` is also saved in Multisig.
+4. Default Response is returned.
+5. If a newer `WorkerSet` was found, a `TransferOperatorship` command is added to the batch. The new `WorkerSet` is stored as the next `WorkerSet`.
+6. The Multisig contract is called asking to sign the binary message
+7. Multisig emits event `SigningStarted` indicating a new multisig session has started
+8. Multisig triggers a reply in Prover returning the newly created session ID which is then stored with the batch for reference
+9. Prover contract emits event `ProofUnderConstruction` which includes the ID of the proof being constructed.
+10. Signers submit their signatures until threshold is reached
+11. Multisig emits event indicating the multisig session has been completed
+12. Relayer queries Prover for the proof, using the proof ID
+13. Prover queries Multisig for the multisig session, using the session ID
+14. Multisig replies with the multisig state, the list of collected signatures so far and the snapshot of participants.
+15. If the Multisig state is `Completed`, the Prover finalizes constructing the proof and returns the `GetProofResponse` struct which includes the proof itself and the data to be sent to the destination gateway. If the state is not completed, the Prover returns the `GetProofResponse` struct with the `status` field set to `Pending`.
+
+
+## ConfirmWorkerSet sequence diagram
+
+```mermaid
+sequenceDiagram
+autonumber
+participant Relayer
+box LightYellow Axelar
+participant Prover
+participant Voting Verifier
+participant Multisig
+end
+Relayer->>+Voting Verifier: ExecuteMsg::ConfirmWorkerSet
+Voting Verifier-->>-Relayer: returns Response
+Relayer->>+Prover: ExecuteMsg::ConfirmWorkerSet
+Prover->>+Voting Verifier:QueryMsg::IsWorkerSetConfirmed
+Voting Verifier-->>-Prover: query result
+Prover->>+Multisig: ExecuteMsg::KeyGen
+Multisig-->>-Prover: returns Response
+Prover-->>-Relayer: returns AppResponse
+```
+
+1. Once the destination gateway emits a `OperatorshipTransferred` picked up by the Relayer, the Relayer calls Voting Verifier to create a poll. 
+2. Default Response is returned.
+3. Once the poll is completed, the Relayer calls the Prover to confirm if the `WorkerSet` was updated.
+4. The Prover queries the Voting Verifier to check if the `WorkerSet` is confirmed.
+5. The Voting Verifier returns if the `WorkerSet` is confirmed. If true, the `Prover` stores the `WorkerSet`.
+6. The new `WorkerSet` is stored in Multisig.
+7. Default response is returned.
+8. AppResponse is returned.
```

Comment:
Can we just have one section combining these two, with a sequence diagram as well as a graph that shows all of the components, including the relayer and the gateway on the external chain?

Reply:
Yes, will do!